### PR TITLE
[VRFv2+] Change request commitment from calldata to memory

### PR DESCRIPTION
**Pull request by vreff**
Should save ~1-1.5k gas

### CODE REVIEW

**Comment by makramkd** [comment]
Superceded by #10171 ?

**Comment by vreff** [comment]
> Superceded by https://github.com/smartcontractkit/chainlink/pull/10171 ?

That PR was needed for this PR, but no this is unique. It actually makes the external function also pass in memory for requestCommitment.

File: contracts/src/v0.8/dev/vrf/VRFCoordinatorV2Plus.sol [hunk 1]
```diff
@@ -419,7 +419,7 @@ contract VRFCoordinatorV2Plus is VRF, SubscriptionAPI {
 
     // stack too deep error
     {
-      bool nativePayment = _fromBytes(rc.extraArgs).nativePayment;
+      bool nativePayment = uint8(rc.extraArgs[rc.extraArgs.length - 1]) == 1;
```
[the remark this answers is not in the document]
Comment (makramkd) [reply contracts/src/v0.8/dev/vrf/VRFCoordinatorV2Plus.sol:422]: I guess this is okay because the `extraArgs` at this point should be well formed.

**Review by makramkd — DISMISSED** [review]

File: contracts/src/v0.8/dev/vrf/VRFCoordinatorV2Plus.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (vreff) [reply contracts/src/v0.8/dev/vrf/VRFCoordinatorV2Plus.sol:422]: Yep. Have to do something a bit more low-level.

[the remark this answers is not in the document]
Comment (vreff) [reply contracts/src/v0.8/dev/vrf/VRFCoordinatorV2Plus.sol:422]: And yes, we already ensure in the request that this object is well-formed.

**Review by jinhoonbang — APPROVED** [review]

**Pull request merged**